### PR TITLE
Extra  tests + UrlEncode handle null

### PR DESCRIPTION
**Pull request by 304NotModified**
This change is [Reviewable](https://reviewable.io/reviews/nlog/nlog/1639)

### CODE REVIEW

File: src/NLog/Internal/UrlHelper.cs [hunk 1]
```diff
@@ -51,6 +51,8 @@ internal class UrlHelper
         /// <returns>Encoded url.</returns>
         internal static string UrlEncode(string str, bool spaceAsPlus)
         {
+            if (str == null) return string.Empty;
```
Comment (304NotModified) [inline src/NLog/Internal/UrlHelper.cs]: test
